docker-compose.yml

```diff
@@ -9,6 +9,8 @@ services:
       context: ./
       dockerfile: Dockerfile
       target: bookbrainz-dev
+      args:
+        - GIT_COMMIT_SHA=${GIT_COMMIT_SHA}
```

Comment:
Maybe this change is useless, you can already pass this argument directly to the docker-compose command.
Originally I had also included the first example command from the commit message in `develop.sh`, but I have decided to drop this again, because this would lead to the entire image being built from scratch each time `develop.sh`  is run for a different commit.

Reply:
The more I think about this, the more I'm determined to drop the commit with this change...
It also results in a `WARN[0000] The "GIT_COMMIT_SHA" variable is not set. Defaulting to a blank string.` Docker warning because I intentionally want to have no fallback at this level in order to be able to handle the different fallback cases separately:
1) Display the repository link in the footer as "unknown revision" but ...
1) ... append "master" to the GH tree base URL to still have a valid link.
1) Fail to extract a version number and fallback to "latest".
    Thinking about this one again, it might make more sense to fallback to "dev" here, as it is used in the user agent and could be any checked out commit. What do you think?

Reply:
There are two cases where I want the version extraction to fail deliberately:
1. GIT_COMMIT_SHA is not set inside the container. This happens in development unless it is explicitly set as Compose argument or as a local environment variable (with this patch).
2. GIT_COMMIT_SHA contains a real commit hash and not a version tag which can be parsed. This could happen depending on how the production image is built.

In both of these cases we agreed that the user agent should fall back to "dev" in absence of a version number.
But I explicitly did not want to set the fallback in docker-compose already as that would also cause the GIT_COMMIT_SHA fallbacks for the footer (https://github.com/metabrainz/bookbrainz-site/pull/943/commits/871ff902e5488cc4837192b26c4ce01a0d856b2a) to be ignored and lead to a broken link https://github.com/metabrainz/bookbrainz-site/tree/dev.

As I already mentioned above, the standard way to set GIT_COMMIT_SHA is via the Compose args, which is done when building a production image.
So my remaining question is whether I should drop the possibility to also set this via a local environment variable (https://github.com/metabrainz/bookbrainz-site/pull/943/commits/3a046a2003302af259120e9d0f8f6950f62f43f6)? I would say yes per my comments above.

Reply:
Commit has been dropped, I've force-pushed the branch.